# This config file is used by ros2_control
/**:
  controller_manager:
    ros__parameters:
      update_rate: 400  # Hz
      thread_priority: 99

      joint_state_broadcaster:
        type: joint_state_broadcaster/JointStateBroadcaster

      feedback_controller:
        type: feedback_controller/FeedbackController

      # spring_actuator_controller:
      #   type: spring_actuator_controller/SpringActuatorController

      # joint_trajectory_command_broadcaster:
      #   type: joint_trajectory_command_broadcaster/JointTrajectoryCommandBroadcaster

/**:
  feedback_controller:
    ros__parameters:
      joints:
        - robot1/joint1
        - robot1/joint2
        - robot1/joint3
        - robot1/joint4
        - robot1/joint5
        - robot1/joint6
        - robot1/joint7
      command_interfaces:
        - effort
      state_interfaces:
        - position
        - velocity

# /**:
#   spring_actuator_controller:
#     ros__parameters:
#       joints:
#         - rh_r1_joint
#       command_interfaces:
#         - effort
#       state_interfaces:
#         - position
#         - velocity

# /**:
#   joint_trajectory_command_broadcaster:
#     ros__parameters:
#       joints:
#         - joint1
#         - joint2
#         - joint3
#         - joint4
#         - joint5
#         - joint6
#         - rh_r1_joint
#       interfaces:
#         - position
#       reverse_joints:
#         - rh_r1_joint
#       offsets:
#         - 0.0
#         - 0.0
#         - 0.0
#         - 0.0
#         - 0.0
#         - 0.0
#         - 0.2